model/utils_test.go

```diff
@@ -736,3 +737,32 @@ func checkNowhereNil(t *testing.T, name string, value interface{}) bool {
 		return true
 	}
 }
+
+func TestSanitizeUnicode(t *testing.T) {
+	buf := bytes.Buffer{}
+	buf.WriteString("Hello")
+	buf.WriteRune(0x1d173)
+	buf.WriteRune(0x1d17a)
+	buf.WriteString(" there.")
+
+	musicArg := buf.String()
+	musicWant := "Hello there."
+
+	tests := []struct {
+		name string
+		arg  string
+		want string
+	}{
+		{name: "empty string", arg: "", want: ""},
+		{name: "ascii only", arg: "Hello There", want: "Hello There"},
+		{name: "allowed unicode", arg: "Ādam likes Iñtërnâtiônàližætiøn", want: "Ādam likes Iñtërnâtiônàližætiøn"},
+		{name: "blacklist char, don't reverse string", arg: "\u202E2resu", want: "2resu"},
```

Comment:
Should we add a allowed `unicode` escaped character to the test cases?

Reply:
Ok, I added that.  It's really just testing the compiler's interpretation of different unicode syntax.  https://github.com/mattermost/mattermost-server/pull/14214/commits/9546f681cccc6783e35f30b204ddf5c887b24951